src/main/java/org/opensearch/ad/transport/handler/AnomalyIndexHandler.java

```diff
@@ -212,12 +213,12 @@ void saveIteration(IndexRequest indexRequest, String detectorId, Iterator<TimeVa
                             } else {
                                 TimeValue nextDelay = backoff.next();
                                 LOG.warn(String.format(Locale.ROOT, RETRY_SAVING_ERR_MSG, detectorId), cause);
-                                // copy original request's source without other information like autoGeneratedTimestamp
-                                // otherwise, an exception will be thrown indicating autoGeneratedTimestamp should not be set
-                                // while request id is already set (id is set because we have already sent the request before).
-                                IndexRequest newReuqest = new IndexRequest(indexRequest.index());
-                                newReuqest.source(indexRequest.source(), indexRequest.getContentType());
-                                threadPool.schedule(() -> saveIteration(newReuqest, detectorId, backoff), nextDelay, ThreadPool.Names.SAME);
+                                threadPool
```

Comment:
Why remove the comment? It explains why save iteration is using a clone instead of the index request

Reply:
I moved the comments to BulkUtil.cloneIndexRequest as that's the low-level detail of the clone method.